flutter-for-xamarin-forms.md

```diff
@@ -0,0 +1,2322 @@
+* `resumed` — The application is visible and responding to user input.
+* `suspending` — The application is suspended momentarily. The iOS platform
+has no equivalent event.
+
+For more details on the meaning of these states, see
+[`AppLifecycleStatus` documentation](https://docs.flutter.io/flutter/dart-ui
+/AppLifecycleState-class.html).
+
+# Layouts
+
+## What is the equivalent of a StackLayout?
+
+In Xamarin.Forms you can create a `StackLayout` with an `Orientation` of Horizontal or Vertical.
+Flutter has a similar approach, however you would use the `Row` or `Column` widgets.
+
+If you notice the two code samples are identical with the exception of the
+"Row" and "Column" widget. The children are the same and this feature can be
+exploited to develop rich layouts that can change overtime with the same
+children.
+
+<!-- skip -->
+{% prettify dart %}
+  @override
+  Widget build(BuildContext context) {
+    return new Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: <Widget>[
+        new Text('Row One'),
+        new Text('Row Two'),
+        new Text('Row Three'),
+        new Text('Row Four'),
+      ],
+    );
+  }
+{% endprettify %}
+
+<!-- skip -->
+{% prettify dart %}
+  @override
+  Widget build(BuildContext context) {
+    return new Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: <Widget>[
+        new Text('Column One'),
+        new Text('Column Two'),
+        new Text('Column Three'),
+        new Text('Column Four'),
+      ],
+    );
+  }
+{% endprettify %}
+
+## What is the equivalent of a Grid?
+
+The closest equivalent of a `Grid` would be to use a `GridView`. This is much more powerful
+than what you are used to in Xamarin.Forms. A `GridView` contains a scrolling as well.
+
+<!-- skip -->
+{% prettify dart %}
+  GridView.count(
+    // Create a grid with 2 columns. If you change the scrollDirection to 
+    // horizontal, this would produce 2 rows.
+    crossAxisCount: 2,
+    // Generate 100 Widgets that display their index in the List
+    children: List.generate(100, (index) {
+      return Center(
+        child: Text(
+          'Item $index',
+          style: Theme.of(context).textTheme.headline,
+        ),
+      );
+    }),
+  );
+{% endprettify %}
+
+Alternatively you might have used `Grid`s in Xamarin.Forms to overlay elements. In this case,
+you would use a `Stack` to overlay elements on top of each other.
+
+This sample will create two icons that overlap each other.
+
+<!-- skip -->
+{% prettify dart %}
+  child: new Stack(
+    children: <Widget>[
+      new Icon(Icons.add_box, size: 24.0, color: const Color.fromRGBO(0,0,0,1.0)),
+      new Positioned(
+        left: 10.0,
+        child: new Icon(Icons.add_circle, size: 24.0, color: const Color.fromRGBO(0,0,0,1.0)),
+      ),
+    ],
+  ),
+{% endprettify %}
+
+## What is the equivalent of a ScrollView?
+
+In Xamarin.Forms you use a ScrollView wrap around a `VisualElement` and if the content is

```

Comment:
In Xamarin.Forms, a ScrollView wraps around a `VisualElement` and, if the content is larger than the device screen, it scrolls.